### PR TITLE
[Notifications P1] Comment Moderation View

### PR DESCRIPTION
**Pull request by alpavanoglu**
Refs #https://github.com/Automattic/wordpress-mobile/issues/33

## Description
This PR adds a customisable moderation view for all moderation states. This component will be used in the following Comment Moderation Tickets [#https://github.com/Automattic/wordpress-mobile/issues/33, #https://github.com/Automattic/wordpress-mobile/issues/34, #https://github.com/Automattic/wordpress-mobile/issues/35, #https://github.com/Automattic/wordpress-mobile/issues/36]

## Testing Steps
Actions are not yet implemented other than causing the state & UI to change. ViewModel is in place to allow further implementation. UI is not attached to the flow so testing is possible via Previews.

Go to `CommentModerationView.swift` file and enable canvas (comment out `didFinishLaunchingWithOptions` and `willFinishLaunchingWithOptions` functions in `WordPressAppDelegate` and select `WordPress` as build target instead of `Jetpack`). Scrolling horizontally will reveal all possible states. Each one is interactive. Verify if the designs match to yWt5gg3nWORhu079Qfv3mS-fi-1118_2402 (the button colors are updated to sync with Design System Component).

## Screenshot Reference
The tertiary button is blue instead of Jetpack green due to running `WordPress` build for Previews. The dyanmic Design System color is used.

https://github.com/wordpress-mobile/WordPress-iOS/assets/15968946/304b5757-55db-4ac3-9beb-ab59c1130b5a

## Regression Notes
1. Potential unintended areas of impact

2. What I did to test those areas of impact (or what existing automated tests I relied on)

3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:

- [ ] I have completed the Regression Notes.
- [ ] I have considered adding unit tests for my changes.
- [ ] I have considered adding accessibility improvements for my changes.
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Testing checklist:
- [ ] WordPress.com sites and self-hosted Jetpack sites.
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 500 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WordPress/Classes/ViewRelated/Notifications/Comment Moderation/CommentModerationView.swift [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+    let content: T
+
+    private var transition: AnyTransition = .opacity
+
+    private let animationDuration = 0.25
+
+    private let insertionAnimationDelay = 0.8
+
+    private var insertionAnimation: Animation {
+        return removalAnimation.delay(insertionAnimationDelay)
```
Comment (salimbraksa) [inline WordPress/Classes/ViewRelated/Notifications/Comment Moderation/CommentModerationView.swift:46]: Hey @alpavanoglu There was a small regression when I pushed this commit https://github.com/wordpress-mobile/WordPress-iOS/pull/23127/commits/fc2c17644f4e5ac1406b0ccb6204b1f5f1a92aa4

```suggestion
        return removalAnimation.delay(animationDuration * insertionAnimationDelay)
```

File: WordPress/Classes/ViewRelated/Notifications/Comment Moderation/CommentModerationView.swift [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+
+    let title: String
+    let icon: V
+    let content: T
+
+    private var transition: AnyTransition = .opacity
+
+    private let animationDuration = 0.25
+
+    private let insertionAnimationDelay = 0.8
```
Comment (salimbraksa) [inline WordPress/Classes/ViewRelated/Notifications/Comment Moderation/CommentModerationView.swift]: Hey @alpavanoglu There was a small regression when I pushed this commit https://github.com/wordpress-mobile/WordPress-iOS/pull/23127/commits/fc2c17644f4e5ac1406b0ccb6204b1f5f1a92aa4

```suggestion
    private var insertionAnimationDelay: TimeInterval {
        return animationDuration * 0.8
    }
```

Comment (alpavanoglu) [reply WordPress/Classes/ViewRelated/Notifications/Comment Moderation/CommentModerationView.swift]: Resolved in: [e1b1031](https://github.com/wordpress-mobile/WordPress-iOS/pull/23127/commits/e1b1031f7127a9c67d4a18ef1a825a8a4bda7b05)

LMK if it is fine now. If so we can merge

**Review by salimbraksa — APPROVED** [review]
LGTM! 🚀

**Pull request merged**